docs/concepts/primer.rst

```diff
@@ -234,29 +235,79 @@ TODO: finish section
 Native Interoperability
 ^^^^^^^^^^^^^^^^^^^^^^^
 
-Every operating system in current use provides a lot of platform support for 
-various programming tasks. .NET provides several ways to tap into those APIs. 
-Collectively, this support is called "native interoperability" and in this 
-section we will take a look at how to access native APIs from managed, .NET 
-code. 
+Every operating system in current use provides a lot of platform support for
+various programming tasks. .NET provides several ways to tap into those APIs.
+Collectively, this support is called "native interoperability" and in this
+section we will take a look at how to access native APIs from managed, .NET
+code.
 
-The main way to do native interoperability is via "platform invoke" or P/Invoke 
-for short. This support in .NET Core is available across Linux and Windows 
-platforms. Another, Windows-only way of doing native interoperability is known 
-as "COM interop". It's main goal is to allow using 
+The main way to do native interoperability is via "platform invoke" or P/Invoke
+for short. This support in .NET Core is available across Linux and Windows
+platforms. Another, Windows-only way of doing native interoperability is known
+as "COM interop". It's main goal is to allow using
 `COM components <https://msdn.microsoft.com/en-us/library/bwa2bx93.aspx>`_
-in managed code. It is built on top of P/Invoke infrastructure, but it works in 
-subtly different ways. 
+in managed code. It is built on top of P/Invoke infrastructure, but it works in
+subtly different ways.
 
-Most of Mono's (and thus Xamarin's) interoperability support for Java and 
-Objective-C are built similarly, that is, they use the same principles. 
+Most of Mono's (and thus Xamarin's) interoperability support for Java and
+Objective-C are built similarly, that is, they use the same principles.
 
-Read more about it in the :doc:`native-interop` document. 
+Read more about it in the :doc:`native-interop` document.
+
+Unsafe Code
+^^^^^^^^^^^
+
+The CLR enables the ability to access native memory and do pointer
+arithmetic via ``unsafe`` code. These operations are needed for certain algorithms and system interoperability.  Although powerful, use of unsafe code is discouraged unless it is necessary to interop with system APIs or implement the most efficient algorithm.  Unsafe code may not predictably run in all environments, and also loses the benefits of a garbage collector and type safety.  It's recommended to confine unsafe code as much as possible, and test that code thoroughly.
```

Comment:
I'm still wondering what you mean by "may not predictably run in all environments".